test/extended/olm/olmv1.go

```diff
@@ -417,3 +501,318 @@ func checkFeatureCapability(oc *exutil.CLI) {
+	jobName := fmt.Sprintf("catalog-server-load-test-%s", rand.String(5))
+	namespace := "default"
+
+	baseURL, err := oc.AsAdmin().WithoutNamespace().Run("get").Args(
+		"clustercatalogs.olm.operatorframework.io",
+		catalog,
+		"-o=jsonpath={.status.urls.base}").Output()
+	o.Expect(err).NotTo(o.HaveOccurred())
+	o.Expect(baseURL).NotTo(o.BeEmpty(), fmt.Sprintf("base URL not found for catalog %s", catalog))
+
+	var serviceURL string
+	if query != "" {
+		serviceURL = fmt.Sprintf("%s/api/v1/%s?%s", baseURL, endpoint, query)
+	} else {
+		serviceURL = fmt.Sprintf("%s/api/v1/%s", baseURL, endpoint)
+	}
+	g.GinkgoLogr.Info(fmt.Sprintf("Using service URL: %s", serviceURL))
+
+	g.By(fmt.Sprintf("Load testing /api/v1/%s endpoint for %s", endpoint, catalog))
+
+	jobYAML := fmt.Sprintf(`
+apiVersion: batch/v1
+kind: Job
+metadata:
+  name: %s
+  namespace: %s
+spec:
+  backoffLimit: 2
+  template:
+    spec:
+      containers:
+      - name: wrk2-tester
+        image: registry.redhat.io/ubi8/ubi:latest
+        command: 
+        - /bin/bash
+        - -c
+        - |
+          set -ex
+          # Set test parameters
+          SERVICE_URL="%s"
+          CONCURRENT_CONNECTIONS=10
+          TEST_DURATION_SECONDS=30
+          REQUEST_RATE=100
+          MAX_LATENCY_THRESHOLD_MS=%d
+          SUCCESS_RATE_THRESHOLD=%.1f  
+          # Install required packages
+          dnf install -y git gcc openssl-devel zlib-devel make
+          # Clone and build wrk2
+          git clone https://github.com/giltene/wrk2.git
+          cd wrk2
+          make
+          # Create a Lua script processing results
+          cat > print_report.lua << 'EOL'
+          
+          function done(summary, latency, requests)
+              -- Print basic stats
+              io.write("\n\n=== LOAD TEST RESULTS ===\n")
+              io.write(string.format("Total requests: %%d\n", summary.requests))
+              io.write(string.format("Socket errors: connect %%d, read %%d, write %%d, timeout %%d\n",
+                  summary.errors.connect, summary.errors.read, summary.errors.write, summary.errors.timeout))
+              io.write(string.format("HTTP errors: %%d\n", summary.errors.status))
+              io.write(string.format("Request rate: %%.2f requests/s\n", summary.requests / summary.duration * 1000000))
+              
+              -- Calculate success rate
+              local total_errors = summary.errors.status + summary.errors.connect + 
+                                  summary.errors.read + summary.errors.write + summary.errors.timeout
+              local success_count = summary.requests - total_errors
+              local success_rate = success_count / summary.requests * 100
+              io.write(string.format("Success rate: %%.2f%%%%\n", success_rate))
+              
+              
+              -- Check if we met our success rate threshold
+              local success_threshold = tonumber(os.getenv("SUCCESS_RATE_THRESHOLD") or 90)
```

Comment:
Any reason we need the `or 90` here? If we do need the fallback default, I'd recommend `or 100`, I think?

Reply:
This has all been changed to check for 100% success rate.